frontend/src/lib/components/HTMLElementsDisplay/HTMLElementsDisplay.tsx

```diff
@@ -114,16 +144,29 @@ export function HTMLElementsDisplay({
                 </LemonBanner>
             )}
             <div className="px-4 rounded bg-default">
-                {elements.length ? (
+                {parsedElements.length ? (
                     <>
-                        <Tags
-                            elements={elements}
-                            highlight={highlight}
-                            editable={editable}
-                            parsedCSSSelectors={parsedSelectors}
-                            onChange={(index, s) => setParsedSelectors({ ...parsedSelectors, [index]: s })}
-                        />
-                        <CloseAllTags elements={elements} />
+                        {elementsToShowDepth ? (
+                            <pre
+                                className="p-1 m-0 opacity-50 text-white text-sm cursor-pointer"
+                                data-attr="elements-display-show-more-of-chain"
+                                onClick={showAdditionalElements}
+                            >
+                                {`Show ${Math.min(3, elementsToShowDepth)} more parent${
+                                    elementsToShowDepth > 1 && 's'
+                                } (${elementsToShowDepth} hidden)`}
+                            </pre>
+                        ) : null}
+                        <Fade visible={true} style={{ position: 'static' }}>
+                            <Tags
+                                elements={parsedElements}
+                                highlight={highlight}
+                                editable={editable}
+                                parsedCSSSelectors={parsedSelectors}
+                                onChange={(index, s) => setParsedSelectors({ ...parsedSelectors, [index]: s })}
+                            />
+                            <CloseAllTags elements={parsedElements} />
+                        </Fade>
```

Comment:
Adding a Fade tag here isn't necessary. You already have it under `Tags` and `CloseAllTags`.

Reply:
Resolved in ad73878